### PR TITLE
[test] - sidechain test coverage

### PR DESCRIPTION
**Pull request by 0xahtle7**
- Refactors sidechain deployment scripts 
- Adds sidechain deployment mocks 
- Adds sidechain deployment phases 
- Adds VoterProxyLite, BoosterLite, BoosterOwnerLite test coverage
- Adds Create2Factory , BridgeDelegateReceiver test coverage

### CODE REVIEW

File: scripts/deploySidechain.ts [hunk 1]
```diff
@@ -73,12 +73,15 @@ export async function deployCanonicalPhase(
...
         new L1Coordinator__factory(deployer),
         "L1Coordinator",
-        [config.lzEndpoint, phase6.booster.address, config.token, phase2.cvx.address, auraProxyOFT.address],
+        [config.l1LzEndpoint, phase6.booster.address, config.token, phase2.cvx.address, auraProxyOFT.address],
         {},
         debug,
         waitForBlocks,
     );
 
+    // const tx = await l1Coordinator.transferOwnership(multisigs.daoMultisig);
```
Comment (0xahtle7) [inline scripts/deploySidechain.ts:82]: @phijfry  transfer ownship to avoid forgetting doing that in the future.

File: scripts/deploySidechain.ts [hunk 2]
```diff
@@ -262,18 +290,45 @@ export async function deploySidechainSystem(
...
+    // Configure L1 ,L2 Communications
+    if (l1Configuration) {
+        const { addresses: l1ExtConfig, canonical } = l1Configuration;
+        await setTrustedRemoteSidechain(canonical, contracts, l1ExtConfig.canonicalChainId, debug, waitForBlocks);
+    }
+
+    tx = await l2Coordinator.initialize(booster.address, extConfig.token);
+    await waitForTx(tx, debug, waitForBlocks);
+
+    // TODO @phijfry confirm this
```
Comment (0xahtle7) [inline scripts/deploySidechain.ts:327]: @phijfry  transfer ownship to avoid forgetting doing that in the future.

File: scripts/deploySidechain.ts [hunk 3]
```diff
@@ -52,18 +46,24 @@ export interface CanonicalPhaseDeployed {
...
+    //         Deployer : l1Coordinator.transferOwnership(multisigs.daoMultisig);
+    //     Protocol DAO : l1Booster.setBridgeDelegate(l1Coordinator.address);
+    // -----------------------------
+
     const auraProxyOFT = await deployContract<AuraProxyOFT>(
         hre,
         new AuraProxyOFT__factory(deployer),
         "AuraProxyOFT",
-        [config.lzEndpoint, phase2.cvx.address, phase2.cvxLocker.address],
+        [config.l1LzEndpoint, phase2.cvx.address, phase2.cvxLocker.address],
```
Comment (phijfry) [inline scripts/deploySidechain.ts:66]: just call these lzEndpoint,  no need to change them to `L1`, `L2` as that comes from the context of the config.

File: scripts/deploySidechain.ts (hunk 1 above)
Comment (phijfry) [reply scripts/deploySidechain.ts:82]: Yes that is fine.

Comment (phijfry) [reply scripts/deploySidechain.ts:82]: Just make sure tests still work

File: scripts/deploySidechain.ts [hunk 4]
```diff
@@ -166,19 +194,19 @@ export async function deploySidechainSystem(
...
+    const auraOFTTransferOwnership = AuraOFT__factory.createInterface().encodeFunctionData("transferOwnership", [
         deployerAddress,
     ]);
     const auraOFT = await deployContractWithCreate2<AuraOFT, AuraOFT__factory>(
         hre,
         create2Factory,
         new AuraOFT__factory(deployer),
         "AuraOFT",
-        [naming.coordinatorName, naming.coordinatorSymbol, addresses.lzEndpoint, extConfig.canonicalChainId],
+        [naming.coordinatorName, naming.coordinatorSymbol, extConfig.l2LzEndpoint, extConfig.canonicalChainId],
```
Comment (phijfry) [inline scripts/deploySidechain.ts:205]: Same comment as above

File: scripts/deploySidechain.ts [hunk 5]
```diff
@@ -262,18 +290,45 @@ export async function deploySidechainSystem(
...
+            proxyFactory,
+        },
+        poolManager,
+        auraOFT,
+        l2Coordinator,
+    };
+
     let tx: ContractTransaction;
+    // Configure L1 ,L2 Communications
+    if (l1Configuration) {
```
Comment (phijfry) [inline scripts/deploySidechain.ts:319]: If you want to do this here then make it required, the L1 will always be deployed before the L2 so it should always exist.

File: scripts/deploySidechain.ts (hunk 2 above)
Comment (phijfry) [reply scripts/deploySidechain.ts:327]: lgtm

File: scripts/deploySidechain.ts [hunk 6]
```diff
@@ -300,88 +355,52 @@ export async function deploySidechainSystem(
...
-    token: MockERC20;
-    bpt: MockBalancerPoolToken;
-    minter: MockCurveMinter;
-    gauge: MockCurveGauge;
+    return contracts;
 }
 
-export async function deploySidechainMocks(
-    hre: HardhatRuntimeEnvironment,
+export async function deploySidechainPhase2(
```
Comment (phijfry) [inline scripts/deploySidechain.ts:367]: There is only 1 phase currently. No need to add this as there is no gauge controller etc on the L2s.

File: scripts/deploySidechain.ts [hunk 7]
```diff
@@ -300,88 +355,52 @@ export async function deploySidechainSystem(
...
+    //          Protocol DAO : l1Coordinator.setTrustedRemote(L2_CHAIN_ID, [l2Coordinator.address, l1Coordinator.address])
+    //          Protocol DAO : auraProxyOFT.setTrustedRemote(L2_CHAIN_ID, [auraOFT.address, auraProxyOFT.address])
+    // Pre-2: L2
+    //          Protocol DAO : l1Coordinator.setTrustedRemote(L2_CHAIN_ID, [l2Coordinator.address, l1Coordinator.address])
+    //          Protocol DAO : auraProxyOFT.setTrustedRemote(L2_CHAIN_ID, [auraOFT.address, auraProxyOFT.address])
+    // -----------------------------
+    // 2. Sidechain Add pools at PoolManager:
+    // -----------------------------
+    // -----------------------------
+    // POST-2: TreasuryDAO: LBP.updateWeightsGradually
```
Comment (phijfry) [inline scripts/deploySidechain.ts:386]: This function should be removed but these comments aren't right either.

File: scripts/deploySidechainMocks.ts [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        const gauge = await deployContract<MockCurveGauge>(
+            hre,
+            new MockCurveGauge__factory(deployer),
+            "MockCurveGauge",
+            [`TestGauge_${i + 1}`, `tstGauge_${i + 1}`, lptoken.address, []],
+            {},
+            debug,
+        );
+
+        const tx = await voting.vote_for_gauge_weights(gauge.address, 1);
```
Comment (phijfry) [inline scripts/deploySidechainMocks.ts:118]: This is confusing, there is no gauge controller on the L2, this entire mock can just be the simplified thing it was before.

File: scripts/deploySidechainMocks.ts [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+        crvMinter,
+        voting,
+        gauges,
+        crvBpt,
+        lptoken,
+        create2Factory,
+        l2LzEndpoint,
+        addresses: {
+            canonicalChainId: L1_CHAIN_ID,
+            remoteLzChainId: L2_CHAIN_ID,
```
Comment (phijfry) [inline scripts/deploySidechainMocks.ts:145]: The remote chain for the sidechain is the canonical chain. "remote" "local" are directional "sidechain" "canonical" and "l1" "l2" are not directional and should be used instead. I tend to use "sidechain" and "canonical" as easier to read when debugging tests.

File: scripts/deploySystem.ts [hunk 10]
```diff
@@ -167,7 +167,8 @@ interface ExtSystemConfig {
     uniswapRouter?: string;
     sushiswapRouter?: string;
     auraBalGauge?: string;
-    lzEndpoint?: string;
+    l1LzEndpoint?: string;
```
Comment (phijfry) [inline scripts/deploySystem.ts:170]: Same as previous comments, lzEndpoint is fine.

File: tasks/deploy/mainnet-config.ts [hunk 11]
```diff
@@ -153,7 +156,8 @@ const addresses: ExtSystemConfig = {
         ],
         assetsIn: ["0xA13a9247ea42D743238089903570127DdA72fE44", "0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0"],
     },
-    lzEndpoint: "0x66A71Dcef29A0fFBDBE3c6a460a3B5BC225Cd675",
+    canonicalChainId: 101,
```
Comment (phijfry) [inline tasks/deploy/mainnet-config.ts:159]: Similar comment regarding the l1LzEndpoint in that this can just be called `lzChainId` as it's already in the context of the mainnet config.

File: tasks/deploy/sidechain.ts [hunk 12]
```diff
@@ -79,25 +79,29 @@ task("deploy:sidechain:L1")
...
-        const result = await deployCanonicalPhase(hre, config.addresses, phase2, phase6, deployer, debug, tskArgs.wait);
+        const result = await deployCanonicalPhase(hre, deployer, config.addresses, phase2, phase6, debug, tskArgs.wait);
         logContracts(result as unknown as { [key: string]: { address: string } });
     });
 
 task("deploy:sidechain:L2")
     .addParam("wait", "wait for blocks")
     .setAction(async (tskArgs: TaskArguments, hre: HardhatRuntimeEnvironment) => {
         const deployer = await getSigner(hre);
+        const l1Config = canonicalConfigs[hre.network.config.chainId];
```
Comment (phijfry) [inline tasks/deploy/sidechain.ts:90]: This is confusing, you are using the `hre.network.config.chainId` and looking it up on the canonical configs object but this is the L2 deployment task 🤔

Comment (0xahtle7) [reply tasks/deploy/sidechain.ts]: I removed any dependency on l1 for this task

File: scripts/deploySidechainMocks.ts (hunk 8 above)
Comment (0xahtle7) [reply scripts/deploySidechainMocks.ts]: I reverted the changes

File: scripts/deploySidechain.ts (hunk 5 above)
Comment (0xahtle7) [reply scripts/deploySidechain.ts]: I removed it to simplify the script

File: tasks/sidechain/index.ts [hunk 13]
```diff
@@ -144,8 +144,8 @@ task("sidechain:test:send-aura-to-sidechain")
         const remoteConfig = sidechainConfigs[remoteChainId];
         assert(remoteConfig, `Remote config for chain ID ${remoteChainId} not found`);
 
-        const remoteLzChainId = lzChainIds[remoteChainId];
-        assert(remoteLzChainId, "LZ chain ID not found");
+        const sidechainLzChainId = lzChainIds[remoteChainId];
```
Comment (phijfry) [inline tasks/sidechain/index.ts:147]: We can revert this change

File: test-fork/sidechain/FullDeployment.spec.ts [hunk 14]
```diff
@@ -304,11 +303,14 @@ describe("Sidechain", () => {
         });
         it("add trusted remotes to layerzero endpoints", async () => {
             // L1 Stuff
+            canonical.l1Coordinator = canonical.l1Coordinator.connect(dao.signer);
```
Comment (phijfry) [inline test-fork/sidechain/FullDeployment.spec.ts:306]: maybe do this stuff in the test before function?

File: hardhat.config.ts [hunk 15]
```diff
@@ -4,29 +4,15 @@ import "@typechain/hardhat";
...
 import { resolve } from "path";
-
 import { config as dotenvConfig } from "dotenv";
 import { HardhatUserConfig } from "hardhat/config";
+import "./tasks/coverage";
+import { chainIds } from "./tasks/utils/networkAddressFactory";
 
 dotenvConfig({ path: resolve(__dirname, "./.env") });
 
-export const chainIds = {
```
Comment (0xahtle7) [inline hardhat.config.ts:17]: I moved because it is used on some hh tasks , this cause the following error 
```
Error HH9: Error while loading Hardhat's configuration.
```
